api/test/integration/env/configurations/vulnerability/manager/healthcheck/healthcheck.py

```diff
@@ -37,13 +39,30 @@ def get_agents_node(status='active'):
     return {item['id']: item['manager'] for item in response}
 
 
+def get_healthcheck_state():
+    with open(HEALTHCHECK_STATE_FILE, mode='r') as f:
+        return json.loads(f.read())
+
+
+def set_healthcheck_state(state):
+    if state:
+        with open(HEALTHCHECK_STATE_FILE, mode='w') as f:
+            f.write(json.dumps(state, indent=4))
+
+
```

Comment:
Use a file as a flag. No need to save the content of the request. If the file exists, we do not make more requests.

Reply:
Actually I was saving agents on that file to retrieve them later on. I can use this file entirely as a flag but will need to retrieve agents with one more request, if I don't get it wrong. Moreover, If I don't retrieve them from the file, this request will be triggered each time healthcheck runs. I think we should keep it as is but let me know what you think it's our better choice.

Reply:
```python
...
    else:
        # Read state from file instead of api request.
        agents_node = get_healthcheck_state()

    checks_list = list()
    expected_log = "grep -q 'wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment " \
                   "for agent '\\''{}'\\''' /var/ossec/logs/ossec.log"

    if socket.gethostname() == 'wazuh-master':
        checks_list.append(get_manager_health_base())

    for agent in agents_to_check:
        if agents_node[agent] == socket.gethostname():
            # Append only logs that are expected in the current node.
            checks_list.append(check(system(expected_log.format(agent))))

    exit(any(checks_list))
```
Another possible solution could be move this block to fit into else block but `agents_node` still needs to make a request to get agents.

Reply:
I see the point, but the agents' state can change over time. The first request could return less agents than expected due to the `pending` status and this healthcheck would never be `healthy`. It needs to keep making requests until everything is fine.